django/db/backends/mysql/schema.py

```diff
@@ -48,7 +48,11 @@ def _is_limited_data_type(self, field):
         return db_type is not None and db_type.lower() in self.connection._limited_data_types
 
     def skip_default(self, field):
-        return self._is_limited_data_type(field)
+        # MySQL and MariaDB < 10.2.1 don't support defaults for BLOB or TEXT
+        # https://mariadb.com/kb/en/library/blob-and-text-data-types/
+        if not self.connection.mysql_is_mariadb or self.connection.mysql_version < (10, 2, 1):
+            return self._is_limited_data_type(field)
+        return False
```

Comment:
I would revert this, i.e.:
```python
if self.connection.mysql_is_mariadb and self.connection.mysql_version >= (10, 2, 1):
    return True
return self._is_limited_data_type(field)
```

Reply:
I think it's better as-is, that we should consider the happy path to be that defaults don't need skipping, and older versions of MySQL/MariaDB need defaults skipping - especially after https://code.djangoproject.com/ticket/30712